Resources/public/js/views/services/ez-navigationhubviewservice.js

```diff
@@ -53,8 +53,126 @@ YUI.add('ez-navigationhubviewservice', function (Y) {
             });
         },
 
+        _load: function (next) {
+            var api = this.get('capi'),
+                loadError = false,
+                service = this,
+                discoveryService = api.getDiscoveryService(),
+                tasks = new Y.Parallel();
+
+            // Loading root
+            discoveryService.getInfoObject('rootLocation', function (error, response){
+                var rootLocationId;
+
+                if ( error ) {
+                    loadError = true;
+                    return;
+                }
+
+                rootLocationId = response._href;
+
+                service._loadLocationAndContent(rootLocationId, 'rootStruct', tasks.add(function (error) {
+                    var params = {id: '', languageCode: ''};
+
+                    if ( error ) {
+                        loadError = true;
+                        return;
+                    }
+
+                    params.id = service.get('rootStruct').location.get('id');
+                    params.languageCode = service.get('rootStruct').content.get('mainLanguageCode');
+
+                    service.getNavigationItem('content-structure').set(
+                        'route',
+                        {name: 'viewLocation', params: params}
+                    );
+                }));
+            });
+
+            // Loading media root
+            discoveryService.getInfoObject('rootMediaFolder', function (error, response){
+                var rootMediaLocationId;
+
+                if ( error ) {
+                    loadError = true;
+                    return;
+                }
+
+                rootMediaLocationId = response._href;
+
+                service._loadLocationAndContent(rootMediaLocationId, 'rootMediaStruct' , tasks.add(function (error){
+                    var params = {id: '', languageCode: ''};
+
+                    if ( error ) {
+                        loadError = true;
+                        return;
+                    }
+
+                    params.id = service.get('rootMediaStruct').location.get('id');
+                    params.languageCode = service.get('rootMediaStruct').content.get('mainLanguageCode');
+
+                    service.getNavigationItem('content-structure').set(
```

Comment:
`'content-structure'` is not the correct identifier, it should be `'media-library'` (and most likely a test is missing or incorrect